charts/metering-ansible-operator/templates/crds/reportdatasource.crd.yaml

```diff
@@ -39,3 +39,118 @@ spec:
   - name: Age
     type: date
     JSONPath: .metadata.creationTimestamp
+  validation:
+    openAPIV3Schema:
+      type: object
+      required:
+      - spec
+      properties:
+        spec:
+          type: object
+          properties:
+            prometheusMetricsImporter:
+              type: object
+              required:
+              - query
+              # note: we could specify maxProperties: 2 or use `oneOf` to only allow two sub-schemas
+              #       the downside is we would need to update these CRDs if new stuff gets added
+              properties:
+                query:
+                  type: string
+                  minLength: 1
+                storage:
+                  type: object
+                  properties:
+                    storageLocationName:
+                      type: string
+                      minLength: 1
+                prometheusConfig:
+                  type: object
+                  required:
+                  - url
+                  properties:
+                    url:
+                      type: string
+                      format: uri
+            reportQueryView:
+              type: object
+              required:
+              - queryName
+              - inputs
+              properties:
+                queryName:
+                  type: string
+                  minLength: 1
+                inputs:
+                  type: array
+                  minItems: 1
+                  items:
+                    type: object
+                    required:
+                    - name
+                    - type
+                    properties:
+                      name:
+                        type: string
+                        minLength: 1
+                      type:
+                        type: string
+                        minLength: 1
+                        enum:
+                        - varchar
```

Comment:
I don’t think this Is correct. I believe we use string? Iirc it’s based on the go types.